### DISCUSSION TITLE
CoAP set limited to 15 characters

### DISCUSSION THREAD

**Discussion by 97Cweb** [openthread/openthread#10713] [Q&A]
**Describe the bug** A clear and concise description of what the bug is.
When using the CLI commands in ardiuino-esp32, the "coap set" command only accepts 15 characters. I was directed to post here after posting there for it. https://github.com/espressif/arduino-esp32/issues/10350 

**To Reproduce** Information to reproduce the behavior, including:

1. Git commit id Unsure, Arduino-ESP32 somehow inherits this library
2. IEEE 802.15.4 hardware platform ESP-32-C6
3. Build steps use this code and run the commands in "additional Context"
 ```
/*
 * OpenThread.begin(false) will not automatically start a node in a Thread Network
 * The user will need to start it manually using the OpenThread CLI commands
 * Use the Serial Monitor to interact with the OpenThread CLI
 *
 * Type 'help' for a list of commands.
 * Documentation: https://openthread.io/reference/cli/commands
 *
 */

#include "OThreadCLI.h"

void setup() {
  Serial.begin(115200);
  OThreadCLI.begin(false);  // No AutoStart - fresh start
  Serial.println("OpenThread CLI started - type 'help' for a list of commands.");
  OThreadCLI.startConsole(Serial);
}

void loop() {}
```

5. Network topology none required

**Expected behavior** allow more than 15 characters in the set command

**Console/log output** 
```
ot> coap set 0123456789ABCDE
Done

ot> coap set 0123456789ABCDEF
Error 7: InvalidArgs

ot>
```

**Additional context**

coap stop
thread stop
ifconfig down
dataset clear
dataset channel 24
dataset networkkey 00112233445566778899aabbccddeeff
dataset commit active
ifconfig up
thread start
coap start
coap resource json
coap set 0123456789ABCDE
coap set 0123456789ABCDEF

**Comment by jwhui** [comment]
The CoAP resource buffer size is specified here:
https://github.com/openthread/openthread/blob/d7810fa04d46f77eb5b78af81ad0bdb59bfcd436/src/cli/cli_coap.hpp#L80

Feel free to change the buffer size to whatever you need.

Note that the CLI CoAP implementation is intended to serve as a demonstration and not for use with end products.
